src/generator.py

```diff
@@ -0,0 +1,299 @@
+
+
+class FloatGenerator(AbstractGenerator):
+    """
+    Generator for random float values within a specified range.
+    """
+
+    def __init__(self, min_value=0.0, max_value=0.0, precision=2):
+        """
+        Initialize the FloatGenerator with optional parameters.
+
+        Args:
+            min_value (float, optional): The minimum value for the range. Defaults to 0.0.
+            max_value (float, optional): The maximum value for the range. Defaults to 0.0.
+            precision (int, optional): The number of decimal places for the generated values. Defaults to 2.
+        """
+        self.min_value = min_value
+        self.max_value = max_value
+        self.precision = precision
+
+    def generate(self, count: int) -> List[float]:
+        """
+        Generate random float values within the specified range and precision.
+
+        Args:
+            count (int): The number of float values to generate.
+
+        Returns:
+            List[float]: A list of randomly generated float values.
+        """
+        return [round(random.uniform(self.min_value, self.max_value), self.precision) for _ in range(count)]
+
+
+class IntGenerator(AbstractGenerator):
+    """
+    Generator for random integer values within a specified range.
+    """
+
+    def __init__(self, min_value=0, max_value=100):
+        """
+        Initialize the IntGenerator with optional parameters.
+
+        Args:
+            min_value (int, optional): The minimum value for the range. Defaults to 0.
+            max_value (int, optional): The maximum value for the range. Defaults to 100.
+        """
+        self.min_value = min_value
+        self.max_value = max_value
+
+    def generate(self, count: int) -> List[int]:
+        """
+        Generate random integer values within the specified range.
+
+        Args:
+            count (int): The number of integer values to generate.
+
+        Returns:
+            List[int]: A list of randomly generated integer values.
+        """
+        return [random.randint(self.min_value, self.max_value) for _ in range(count)]
+
+
+class LookupGenerator(AbstractGenerator):
+    """
+    Generator for random values from a specified column in a CSV file.
+    """
+
+    def __init__(self, link=None, column=None, **kwargs):
+        """
+        Initialize the LookupGenerator with a link to a CSV file and a column name.
+
+        Args:
+            link (str, optional): The path to the CSV file. Defaults to None.
+            column (str, optional): The column name in the CSV file to use for the value set. Defaults to None.
+        """
+        self.link = link
+        self.column = column
+        # TODO
+        self.dependent_columns = kwargs.get('dependent_column_1', None)
+        self.dependent_concept_id = kwargs.get('dependent_concept_id_1', None)
+        self._load_value_set_from_csv()
+
+    def _load_value_set_from_csv(self):
```

Comment:
What if the file is not found?

Reply:
It was already checked in parser but i add a new check here too